autogpt_platform/backend/backend/integrations/oauth/stripe_link_test.py

```
@@ -0,0 +1,203 @@
+    patcher, _ = mock_http(400, {"error": "slow_down"})
+    with patcher:
+        result = await StripeLinkDeviceAuthHandler().poll_for_tokens("dc")
+    assert result.next_poll_interval == 10
+
+
+@pytest.mark.asyncio
+async def test_poll_raises_on_an_unrecognised_response():
+    """An unknown failure must be loud, not silently treated as pending."""
+    patcher, _ = mock_http(500, {}, text="upstream exploded")
+    with patcher:
+        with pytest.raises(RuntimeError, match="Unexpected response"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)